docs/platform-integration/storage/secure-storage.md

```diff
@@ -0,0 +1,151 @@
+    </application>
+    ```
+
+01. Create a new XML file named _auto_backup_rules.xml_ in the _Resources/xml_ directory with the build action of **AndroidResource**. Set the following content that includes all shared preferences except for `SecureStorage`:
+
+    ```xml
+    <?xml version="1.0" encoding="utf-8"?>
+    <full-backup-content>
+        <include domain="sharedpref" path="."/>
+        <exclude domain="sharedpref" path="${applicationId}.mauiessentials.xml"/>
+    </full-backup-content>
+    ```
+
+# [iOS](#tab/ios)
+
+When developing on the **iOS simulator**, enable the **Keychain** entitlement and add a keychain access group for the application's bundle identifier.
+
+Open the _Entitlements.plist_ in the project and find the **Keychain** entitlement and enable it. This will automatically add the application's identifier as a group.
+
+In the project properties, under **iOS Bundle Signing** set the **Custom Entitlements** to **Entitlements.plist**.
+
+> [!TIP]
+> When deploying to an iOS device, this entitlement isn't required and should be removed.
+
+# [Windows](#tab/windows)
+
+No setup is required.
+
+-----
+<!-- markdownlint-enable MD025 -->
+
+## Use secure storage
+
+The following code examples demonstrate how to use secure storage.
+
+> [!TIP]
+> It's possible that an exception is thrown when calling `GetAsync` or `SetAsync`. This can be caused by a device not supporting secure storage, encryption keys changing, or corruption of data. it's best to handle this by removing and adding the setting back if possible.
+
+### Write a value
+
+To save a value for a given _key_ in secure storage:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_set":::
+
+### Read a value
+
+To retrieve a value from secure storage:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_get":::
+
+> [!TIP]
+> If there isn't a value associated with the key, `GetAsync` returns `null`.
+
+### Remove a value
+
+To remove a specific value, remove the key:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_remove":::
+
+To remove all values, use the `RemoveAll` method:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_remove_all":::
+
+## Platform differences
+
+This section describes the platform-specific differences with the secure storage API.
+
+<!-- markdownlint-disable MD025 -->
+<!-- markdownlint-disable MD024 -->
+# [Android](#tab/android)
+
+The [Android KeyStore](https://developer.android.com/training/articles/keystore.html) is used to store the cipher key used to encrypt the value before it's saved into a [Shared Preferences](https://developer.android.com/training/data-storage/shared-preferences.html) with a filename of _[YOUR-APP-PACKAGE-ID].xamarinessentials_. The value-key used in the shared preferences file is an **MD5** hash of the key passed into the `SecureStorage` APIs.
```

Comment:
Does the filename still have .xamarinessentials appended to it?

Reply:
@davidbritch  No idea. How do you find out?

Reply:
It is $"{ApplicationModel.AppInfo.Current.PackageName}.microsoft.maui.essentials.{feature}"

Reply:
e.g. https://github.com/dotnet/maui/blob/4b833b0cd5f18eab9f3a679c41822010eddff452/src/Essentials/src/Preferences/Preferences.shared.cs#L173

AND

https://github.com/dotnet/maui/blob/4b7dd8acaf2b0040a6a07e71fc140505ee68b7ac/src/Essentials/src/VersionTracking/VersionTracking.shared.cs#L117

Reply:
@jamesmontemagno @davidbritch  ack... seems the "secure storage" feature is using the feature name "preferences" 

https://github.com/dotnet/maui/blob/4b7dd8acaf2b0040a6a07e71fc140505ee68b7ac/src/Essentials/src/SecureStorage/SecureStorage.shared.cs#L102